### ISSUE TITLE
Verify publishing scenarios for satellite resources

### ISSUE THREAD

**Issue by pranavkm** [dotnet/aspnetcore#21379]
Verification for https://github.com/dotnet/aspnetcore/issues/21355

Scenarios to cover (CLI + VS):

* [x] Standalone - @mkArtakMSFT 
* [x] Hosted - @pranavkm 
* [x] Standalone + PWA - @captainsafia 
* [x] Hosted + PWA + Auth - @captainsafia

Things to try out

* [ ] Publish after create
* [ ] Add a resx file to a brand new project, publish
* [ ] Modify a resx file. Verify changes show up.
* [ ] Add a resx file for a new culture after a project has been built. Verify the satellite assembly for the resource shows up

**Comment by SteveSandersonMS** [comment]
Thanks @pranavkm!

Is anyone planning to check on a non-Windows machine? If not that’s fine - I’m happy to check a few cases on Mac/Linux.

Also if we’re including https://github.com/dotnet/aspnetcore/pull/21128 too, let’s make sure we have a build that includes that before verifying. 

No pressure though: if #21128 isn’t going to make it that’s fine with me too.

**Comment by mkArtakMSFT** [comment]
Thanks Steve. It would be great if you could help with validating non-Windows scenarios.
@dotnet/aspnet-blazor-eng folks, please let us know if you intend to validate certain scenarios on non-Windows machines, so that @SteveSandersonMS doesn't duplicate the effort.

**Comment by captainsafia** [comment]
I can validate "Hosted + PWA + Auth" and "Standalone + PWA" on Windows and Linux.

**Comment by mkArtakMSFT** [comment]
Thanks @captainsafia.
I've updated the description to reflect this.

**Comment by SteveSandersonMS** [comment]
Awesome, thanks very much!

**Comment by captainsafia** [comment]
|                                                                                                                           | Hosted + PWA + Auth | Standalone + PWA |
|---------------------------------------------------------------------------------------------------------------------------|---------------------|------------------|
| Publish after create                                                                                                      |                     |                  | 
| Add a resx file to a brand new project, publish                                                                           |                     |                  |  
| Modify a resx file. Verify changes show up.                                                                               |                     |                  |  
| Add a resx file for a new culture after a project has been built. Verify the satellite assembly for the resource shows up |                     |                  |

**Comment by pranavkm** [comment]
Verified a hosted app on Windows. Found one bug not caused by this issue: https://github.com/dotnet/aspnetcore/issues/21406.

**Comment by mkArtakMSFT** [comment]
Verified and eventually everything succeeded (standalone with culture changes ,...)
However, one issue I've faced was some strange linker misbehavior, where the StringLocalizer<Index> constructor was being reported as not found (being stripped out by the linker). I tried to disable linker completely but the behavior remained. However, later when I asked for Pranav to help, I couldn't repro this anymore and things just started to work as expected. Unfortunately, it didn't happen just once - I've been struggling with this for at least 30 minutes before reaching out to Pranav. So I am afraid there is something weird in here we don't have a repro for, which others may face too eventually.
